### PR TITLE
Updates README files

### PR DESCRIPTION
**Pull request by rjwalters**
This cleans up the main repository README and applies a uniform style to the first generation of child crates. README files deeper in the repository might be updated for uniformity in a future PR.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,155 +1,135 @@
...
+| [`ledger-distribution`](./ledger/distribution) | Publishes the ledger to long-term storage. | Full Validator Nodes|
+| [`mobilecoind`](./mobilecoind) | Synchronizes the ledger and provides the desktop API. | Watcher and Validator Nodes |
+
+## Build Instructions
+
+The workspace can be built with `cargo build` and tested with `cargo test`. Either command will recognize the cargo `--release` flag to build with optimizations.
+
+Some crates require additional environment variables to build successfully. For example, the `consensus` crate requires an environment variable "SGX_MODE" to be set for either "hardware mode" or "simulation mode" as well as a variable "IAS_MODE" to indicate the type of attestation service to use. You can find more information in the *BUILD.md* file found in crates that require additional build information.
+
+To simplify the build process, we provide a tool, `mob` which creates a docker container with the required dependencies, including the necessary versions of SGX and protobuf libraries. You can use this tool with the following commands:
```
Comment (sugargoat) [inline README.md:58]:
```suggestion
#### Build Instructions

This workspace is built with `cargo build`, and tested with `cargo test`.

Some crates depend on SGX, which creates build and runtime requirements. (Specifically, `consensus-service`.)

For detailed information about setting up a build environment, how enclaves are built, and configuring
the build, check out [BUILD.md](BUILD.md).

For a quick start, you can build in the same docker image that we use for CI, using the `mob` tool.
This requires you to install docker.

```

Comment (rjwalters) [reply README.md]: updated - take a look.

Comment (sugargoat) [reply README.md]: Perfect - thank you!

File: util/README.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+## util
+
+Testing and bootstrap utilities.
```
Comment (garbageslam) [inline util/README.md:3]: i think this is not quite right, it also contains like `grpc_util`

File: README.md [hunk 3]
```diff
@@ -1,155 +1,134 @@
...
+# At the resulting docker container prompt
 cargo build
 ```
 
-#### License
-
-The components of MobileCoin require different licenses. Look for the LICENSE file in each crate for more information.
-
-#### Overview
+## Overview
```
Comment (garbageslam) [inline README.md:67]: What do you think about moving this out into `OVERVIEW.md` and linking to it from main `README`? Then our README will be more slim

File: README.md [hunk 4]
```diff
@@ -1,155 +1,134 @@
...
+*Full validator nodes* additionally use the `ledger-distribution` target binary to publish a copy of their computed blockchain to content delivery networks (currently to Amazon S3 only). The public blockchain is a zero-knowledge data structure that consists only of *utxos*, *key images* and block metadata used to ensure consistency and to construct Merkle proofs. To build and run ledger distribution, see the [ledger/distribution](./ledger/distribution) crate.
 
-To build and run the MobileCoin Daemon, see the [mobilecoind](./mobilecoind/README.md) crate.
+*Watcher nodes* perform an essential role in the MobileCoin network by verifying the signatures that the *full validator nodes* attach to each block. In this way the *watcher nodes* continuously monitor the integrity of the decentralized MobileCoin network. A *watcher node* also maintains a complete local copy of the blockchain and provide an API for wallet or exchange clients.
 
-### FAQ
+To run a *watcher node*, build and run the [`mobilecoind`](./mobilecoind) daemon.
 
-1. Are my transactions still private in the event of an SGX compromise?
+## FAQ
```
Comment (garbageslam) [inline README.md:101]: What do you think about moving this out to an `FAQ.md`? to make the readme more slim

File: README.md [hunk 5]
```diff
@@ -1,155 +1,134 @@
...
+| [mcbuild](./mcbuild/) | Tools for building and signing enclaves. |
+| [mcconnection](./mcconnection/) | Attested MobileCoin connections. |
+| [mobilecoind](./mobilecoind/) | Blockchain daemon and example client code. |
+| [peers](./peers/) | Peer-to-peer networking. |
+| [sgx](./sgx/) | Support for Intel® Software Guard eXtensions (Intel SGX). |
+| [transaction](./transaction/) | Private transactions. |
+| [util](./util/) | Testing and bootstrap utilities. |
+
+#### Selected Binaries
+| Target | Description | Nodes |
```
Comment (garbageslam) [inline README.md:45]: Maybe rename the column "Nodes" to "Used By" or something?

File: util/README.md (hunk 2 above)
Comment (rjwalters) [reply util/README.md:3]: Can you propose a one sentence descriptor that would be more accurate?

File: README.md (hunk 4 above)
Comment (rjwalters) [reply README.md:101]: We discussed this idea previously and decided to keep the FAQ in the main README.

File: README.md (hunk 3 above)
Comment (rjwalters) [reply README.md:67]: I'm not opposed if a slim README is a goal. I'm also happy with it as it is now.

File: util/README.md (hunk 2 above)
Comment (garbageslam) [reply util/README.md:3]: miscellaneous utilities

Comment (garbageslam) [reply util/README.md:3]: there should really be like, at least two categories: serialization/encodings, and networking

Comment (garbageslam) [reply util/README.md:3]: we dont have to fix it now

Comment (rjwalters) [reply util/README.md]: let's go with misc utils for now

**Pull request merged**